modules/core/image/src/Plugin/ImageEffect/AutoOrientImageEffect.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\farm_image\Plugin\ImageEffect;
+
+use Drupal\Core\File\FileSystemInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Image\ImageInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\image\Attribute\ImageEffect;
+use Drupal\image\ConfigurableImageEffectBase;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Automatically rotates an image based on EXIF Orientation.
+ */
+#[ImageEffect(
+  id: "farm_image_auto_orient",
+  label: new TranslatableMarkup("Auto-orient"),
+  description: new TranslatableMarkup("Automatically rotates an image based on EXIF Orientation.")
```

Comment:
Very minor but these strings could use single quotes to be more consistent?

Reply:
Ah good catch! Fixup: https://github.com/farmOS/farmOS/pull/941/commits/2e9c9788261390d3e58bfece0351d2c2455f4817